src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -965,6 +1104,82 @@ ImmutableOopMapSet* AOTCodeReader::read_oop_map_set() {
   return oopmaps;
 }
 
+#ifndef PRODUCT
+bool AOTCodeCache::write_asm_remarks(CodeBlob& cb) {
+  // Write asm remarks
+  uint* count_ptr = (uint *)reserve_bytes(sizeof(uint));
+  if (count_ptr == nullptr) {
+    return false;
+  }
+  uint count = 0;
+  bool result = cb.asm_remarks().iterate([&] (uint offset, const char* str) -> bool {
+    log_trace(aot, codecache, stubs)("asm remark offset=%d, str='%s'", offset, str);
+    uint n = write_bytes(&offset, sizeof(uint));
+    if (n != sizeof(uint)) {
+      return false;
```

Comment:
Consider using `id_for_C_string()`  and record ID instead of coping string. These strings should be recorded in C strings table already.
If `id_for_C_string()` does not find - assert. We should add `add_C_string()` in missing place.

Reply:
The asm remarks and dbg strings are not currently recorded in C string table.
I tried to add these strings by calling `AOTCodeCache::add_C_string()`  in `AsmRemarkCollection::insert()` but this results in adding LOTS of strings. So I add the strings to the string table only when writing the asm remarks. This keeps the string count in check.